pkg/controller/controller.go

```diff
@@ -575,7 +579,24 @@ func (c *RolloutController) updateStatefulSetPods(ctx context.Context, sts *v1.S
 				continue
 			}
 
-			level.Info(c.logger).Log("msg", fmt.Sprintf("terminating pod %s", pod.Name))
+			// Use the ZPDB to determine if this pod delete is allowed.
+			// The ZPDB serializes requests from this controller and from any incoming voluntary evictions.
+			// For each request, a full set of tests is performed to confirm the state of all pods in the ZPDB scope.
+			// This ensures that if any voluntary evictions have been allowed since this reconcile loop commenced that
+			// the latest information is used to determine the zone/partition disruption level.
+			// If this request returns without error, the pod will be placed into the ZPDB pod eviction cache and will
+			// be considered as not ready until it either restarts or is expired from the cache.
+			err := c.zpdbController.MarkPodAsDeleted(ctx, pod.Namespace, pod.Name, "rollout-controller")
+			if err != nil {
+				// Skip this pod. The reconcile loop regularly runs and this pod will have an opportunity to be re-tried.
+				// Rather than returning false here and abandoning the rest of the updates until the next reconcile,
+				// we allow this update loop to continue. For configurations which have a partition awareness ZPDB it is valid
+				// to have multiple disruptions as long as there is at least one health pod per partition.
+				level.Debug(c.logger).Log("msg", fmt.Sprintf("zpdb denied pod deletion. pod=%s, reason=%s", pod.Name, err.Error()))
```

Comment:
There's no need to format strings like this for log messages, we emit structured key-value pairs:

```suggestion
				level.Debug(c.logger).Log("msg", "zpdb denied pod deletion", "pod", pod.Name, "reason", err)
```